File: app/src/main/java/com/adlab/balda/activities/MainActivity.java
package com.adlab.balda.activities;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import com.adlab.balda.R;
import com.adlab.balda.enums.GameType;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        Button buttonStartOneManGame = findViewById(R.id.b_one_man_game);
        Button buttonStartGameWithAndroid = findViewById(R.id.b_game_with_android);
        Button buttonStartGameWithFriends = findViewById(R.id.b_game_with_friends);
        Button buttonSettings = findViewById(R.id.b_settings);

        buttonStartOneManGame.setOnClickListener(this);
        buttonStartGameWithAndroid.setOnClickListener(this);
        buttonStartGameWithFriends.setOnClickListener(this);
        buttonSettings.setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.b_one_man_game:
                startGameSettingsActivity(GameType.SINGLE);
                break;
            case R.id.b_game_with_android:
                Toast.makeText(this, "Sorry, but game with Android isn't available yet", Toast.LENGTH_SHORT).show();
                break;
            case R.id.b_game_with_friends:
//                Toast.makeText(this, "Sorry, but game with friends isn't available yet", Toast.LENGTH_SHORT).show();
                startGameSettingsActivity(GameType.MULTIPLAYER);
                break;
            case R.id.b_settings:
                Toast.makeText(this, "Sorry, but settings aren't available yet", Toast.LENGTH_SHORT).show();
                break;
        }
    }

    private void startGameSettingsActivity(GameType gameType) {
        startActivity(GameSettingsActivity.createIntent(this, gameType));
    }
}

File: app/src/main/java/com/adlab/balda/activities/MainActivity.kt
package com.adlab.balda.activities

import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.adlab.balda.R
import com.adlab.balda.contracts.MainScreenContract
import com.adlab.balda.databinding.ActivityMainBinding
import com.adlab.balda.enums.GameType
import com.adlab.balda.utils.PresenterManager

class MainActivity: AppCompatActivity(), MainScreenContract.View {

    private lateinit var mPresenter: MainScreenContract.Presenter

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        setSupportActionBar(binding.toolbar)

        binding.bOneManGame.setOnClickListener { mPresenter.startOneManGameClicked() }
        binding.bGameWithAndroid.setOnClickListener {
            Toast.makeText(this, "Sorry, but game with Android isn't available yet", Toast.LENGTH_SHORT).show()
        }
        binding.bGameWithFriends.setOnClickListener { mPresenter.startMultiplayerGameClicked() }
        binding.bSettings.setOnClickListener { mPresenter.settingsClicked() }

        PresenterManager.provideMainScreenPresenter(this)
        mPresenter.start()
    }

    override fun onDestroy() {
        super.onDestroy()
        mPresenter.resetView()
        if (isFinishing) {
            mPresenter.cleanup()
            PresenterManager.resetMainScreenPresenter()
        }
    }

    override fun setPresenter(presenter: MainScreenContract.Presenter) {
        mPresenter = presenter
    }

    override fun showInitiation() {
        binding.initRoot.root.visibility = View.VISIBLE
        binding.bOneManGame.visibility = View.GONE
        binding.bGameWithAndroid.visibility = View.GONE
        binding.bGameWithFriends.visibility = View.GONE
        binding.bSettings.visibility = View.GONE
    }

    override fun showMainButtons() {
        binding.initRoot.root.visibility = View.GONE
        binding.bOneManGame.visibility = View.VISIBLE
        binding.bGameWithAndroid.visibility = View.VISIBLE
        binding.bGameWithFriends.visibility = View.VISIBLE
        binding.bSettings.visibility = View.VISIBLE
    }

    override fun navigateToGameSettings(gameType: GameType) {
        startActivity(GameSettingsActivity.createIntent(this, gameType))
    }

    override fun navigateToSettings() {
        startActivity(RulesActivity.createIntent(this))
    }
}